### PR TITLE
chore(contracts): refactor contract deployments

### PR DESCRIPTION
**Pull request by kevinhalliday**
Refactor contkract deployments.

Deployments for each contract should only be allowed under strict conditions, and their configuration should be static, and in version control.

- move deployment logic to lib/contracts
- introduce create3 and proxyadmin deployments
- remove object oriented AVSDeployer, in favor of functional deployments
- remove unnecessary production functions and configuration from avs lib

task: none

### CODE REVIEW

File: contracts/script/eigen/DeployEigenLayer.s.sol [hunk 1]
```diff
@@ -35,6 +32,11 @@ contract DeployLocalEigenLayer is Script, EigenLayerLocal {
         string memory strategiesJson;
         for (uint256 i = 0; i < deps.strategies.length; i++) {
             IStrategy strat = IStrategy(deps.strategies[i]);
+
+            if (address(strat) == beaconChainETHStrategy) {
+                continue;
+            }
+
```
Comment (kevinhalliday) [inline contracts/script/eigen/DeployEigenLayer.s.sol:39]: `make deploy-eigen-save-anvil-state` was broken

File: contracts/test/avs/common/eigen/EigenLayerFixtures.sol [hunk 2]
```diff
@@ -81,4 +84,20 @@ contract EigenLayerFixtures is Test {
...
+
+        for (uint256 i = 0; i < strats.length; i++) {
+            _strategy[i] = IStrategy(strats[i]);
+            _thirdPartyTransfersForbiddenValues[i] = false;
+        }
+
+        strategyManager.addStrategiesToDepositWhitelist(_strategy, _thirdPartyTransfersForbiddenValues);
+
+        vm.stopPrank();
+    }
```
Comment (kevinhalliday) [inline contracts/test/avs/common/eigen/EigenLayerFixtures.sol:102]: make `deploy-eigen-save-anvil-state` was broken

File: contracts/script/eigen/DeployEigenLayer.s.sol (hunk 1 above)
Comment (pavelnikolov) [reply contracts/script/eigen/DeployEigenLayer.s.sol:39]: Should we add a TODO to revert this later?

Comment (kevinhalliday) [reply contracts/script/eigen/DeployEigenLayer.s.sol:39]: No this fixed it

File: lib/contracts/avs/deploy.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+		_, err = backend.WaitMined(ctx, tx)
+		if err != nil {
+			return common.Address{}, nil, errors.Wrap(err, "wait mined disable allowlist")
+		}
+	}
+
+	tx, err = avs.SetMetadataURI(ownerTxOpts, cfg.MetadataURI)
+	if err != nil {
+		return common.Address{}, nil, errors.Wrap(err, "set metadata uri")
+	}
```
Comment (kevinhalliday) [inline lib/contracts/avs/deploy.go:210]: Might be worth adding metadata uri and and allowlist enabled / disabled to initializer, so we don't need admin txs here. 

And we might have admin controls runnable separately, so we can use this flow (with fireblocks keys) to call admin functions

File: lib/contracts/avs/eigen.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+		return errors.New("avs directory is zero")
+	}
+	if (e.DelegationManager == common.Address{}) {
+		return errors.New("delegation manager is zero")
+	}
+
+	return nil
+}
+
+func holeskeyEigenDeployments() EigenDeployments {
```
Comment (ttarsi) [inline lib/contracts/avs/eigen.go:31]:
```suggestion
func holeskyEigenDeployments() EigenDeployments {
```

**Review by ttarsi — APPROVED** [review]
🚀

File: lib/chainids/chainids.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package chainids
```
[the remark this answers is not in the document]
Comment (kevinhalliday) [reply lib/chainids/chainids.go:1]: Moved all chain ids here, remove arb goerli

**Pull request merged**